android/src/main/java/dev/openfeature/sdk/OpenFeatureAPI.kt

```diff
@@ -1,15 +1,21 @@
 package dev.openfeature.sdk
 
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.SharedFlow
+
 @Suppress("TooManyFunctions")
 object OpenFeatureAPI {
     private var provider: FeatureProvider? = null
     private var context: EvaluationContext? = null
+    private val providersFlow: MutableSharedFlow<FeatureProvider> = MutableSharedFlow(replay = 1)
+    internal val sharedProvidersFlow: SharedFlow<FeatureProvider> get() = providersFlow
 
     var hooks: List<Hook<*>> = listOf()
         private set
 
     fun setProvider(provider: FeatureProvider, initialContext: EvaluationContext? = null) {
         this@OpenFeatureAPI.provider = provider
+        providersFlow.tryEmit(provider)
```

Comment:
Note to self: shall we close the flow on `shutdown`? Shall we emit something else on `clearProvider`?